### PR TITLE
T2B1 wallet creation and seed backup

### PR DESCRIPTION
**Pull request by grdddj**
Fixes https://github.com/trezor/trezor-firmware/issues/3089:
- improve the designs of wallet creation and seed backup flows

### CODE REVIEW

**Comment by Hannsek** [comment]
I've found following inaccuracies: 

image
image
image
image

not centered

align numbers to left side: 
image

Can we get rid of the "processing" text above? 
image

Nitpick: 
It seems to me that the space between the center and right number is bigger then between center and left number.

The carousel is not centerd between the pressed button and header: 
image
image

**Comment by Hannsek** [comment]
image
This screen should look differently.
image

**Comment by grdddj** [comment]
TT screen and number allignment solved in [7772222](https://github.com/trezor/trezor-firmware/pull/3107/commits/7772222e400f3bcb3b5b734eed1ab3a22a66e749)

As for the general centering, I have created a special branch for it - https://github.com/trezor/trezor-firmware/tree/grdddj/tr_content_centering

We are currently having 2px margin below the header, so it is visually delimited from the content in all times. I am deleting this margin there, so some screens change. Let's go through the changes together and decide whether we want it

**Comment by Hannsek** [comment]
image
@andrewkozlik from security pov, should these numbers be shuffled as well when user is choosing the number of shamir shares?

File: core/embed/rust/src/ui/component/text/op.rs [hunk 1]
```diff
@@ -238,7 +241,9 @@ impl<T: StringType + Clone> OpTextLayout<T> {
 #[derive(Clone)]
 pub enum Op<T: StringType> {
     /// Render text with current color and font.
-    Text(T),
+    /// Bool signifies whether the text is complete or not
+    /// (so we know whether to show ellipsis or not).
+    Text((T, bool)),
```
Comment (matejcik) [inline core/embed/rust/src/ui/component/text/op.rs:246]: why not make a separate op for ellipsis instead?

File: core/embed/rust/src/ui/model_tr/layout.rs [hunk 2]
```diff
@@ -761,11 +792,45 @@ extern "C" fn tutorial(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj
...
-                .with_scrollbar(false)
-                .with_common_title("HELLO".into()),
-        )?;
+        let obj = LayoutObj::new(Flow::new(pages).with_scrollbar(false))?;
+        Ok(obj.into())
+    };
+    unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
+}
+
+extern "C" fn new_popup_middle_button(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
```
Comment (matejcik) [inline core/embed/rust/src/ui/model_tr/layout.rs:801]: again, shouldn't "popup middle button" be something like "show_info"?

File: core/embed/rust/src/ui/component/text/op.rs (hunk 1 above)
Comment (grdddj) [reply core/embed/rust/src/ui/component/text/op.rs:246]: How would it then look like? As we cannot know beforehand at which position the ellipsis will be?

To make sure, the use-case of this change is the following:
- 1. we want to show both ellipses (forward and backward one) when we are splitting coherent text (single `Op::Text`, basically one sentence) into more pages

image

- 2. we do not want to show any ellipsis when one coherent text ends at the end of one page and second `Op::Text` (sentence) starts at the beginning of the second page

image

**Comment by andrewkozlik** [comment]
> should these numbers be shuffled as well when user is choosing the number of shamir shares?

I think we don't necessarily need to shuffle these numbers.

File: core/embed/rust/src/ui/component/text/op.rs (hunk 1 above)
Comment (matejcik) [reply core/embed/rust/src/ui/component/text/op.rs:246]: i have a _slightly_ cleaner solution in mind, but it seems to require larger modifications to the Op processing pipeline, for which this is not a good place

for now, just modify the `Text` to not use the tuple. Rename `is_complete` to `continued` and update the comments accordingly, to make it clear that this only happens when it's a `Text` item that is continuing from previous page.

File: core/embed/rust/src/ui/component/text/op.rs [hunk 3]
```diff
@@ -97,14 +97,15 @@ impl<'a, T: StringType + Clone + 'a> OpTextLayout<T> {
...
                 // Drawing text
-                Op::Text(text) => {
+                Op::Text((text, is_complete)) => {
                     // Try to fit text on the current page and if they do not fit,
                     // return the appropriate OutOfBounds message
 
                     // Inserting the ellipsis at the very beginning of the text if needed
                     // (just once for the first Op::Text on the non-first page).
+                    // Also just for incomplete texts that were separated.
                     self.layout.continues_from_prev_page =
```
Comment (matejcik) [inline core/embed/rust/src/ui/component/text/op.rs:107]: we can directly set `self.layout.continues_from_prev_page = continued` (or `!is_complete` in the current version of this code), right? there is no way to get `Text(t, continued=true)` when we already processed some chars

**Comment by grdddj** [comment]
> image should look like this: image
> 
> Why this screen is there? image it should jump right on this screen image
> 
> image -> image
> 
> image -> image this should be the "done" screen image
> 
> image
> 
> BACKUP CHECK instead of SEED CHECK
> 
> image Let's change the font to regular Uppercase "continue".
> 
> image Uppercase "check again".
> 
> image -> image

TR-related things sorted out in https://github.com/trezor/trezor-firmware/pull/3119

For the TT-related things, I think this also deserves its own issue, like "TT recovery flow"

File: core/embed/rust/src/ui/component/text/op.rs (hunk 1 above)
Comment (grdddj) [reply core/embed/rust/src/ui/component/text/op.rs]: Done in [3006d28](https://github.com/trezor/trezor-firmware/pull/3107/commits/3006d285ae8e2eba439680480ffd48cc3b176dae)

**Comment by Hannsek** [comment]
image
This looks strange as I suppose the "new amount" value is on the next screen.

**Comment by grdddj** [comment]
> image This looks strange as I suppose the "new amount" value is on the next screen.

Added an extra newline in https://github.com/trezor/trezor-firmware/pull/3135

image

File: core/embed/rust/src/ui/component/text/op.rs [hunk 4]
```diff
@@ -238,7 +239,9 @@ impl<T: StringType + Clone> OpTextLayout<T> {
 #[derive(Clone)]
 pub enum Op<T: StringType> {
     /// Render text with current color and font.
-    Text(T),
+    /// Bool signifies whether the text continues or not
+    /// (so we know whether to show ellipsis or not).
```
Comment (matejcik) [inline core/embed/rust/src/ui/component/text/op.rs:243]:
```suggestion
    /// Bool signifies whether this is a split Text op continued from previous page.
    /// If true, a leading ellipsis will be rendered.
```

**Comment by grdddj** [comment]
Merged in https://github.com/trezor/trezor-firmware/pull/3135

**Pull request closed without merging**